org.adempiere.base/src/org/compiere/model/MAcctSchema.java

```diff
@@ -179,6 +179,22 @@ public static MAcctSchema getCopy(Properties ctx, int C_AcctSchema_ID, String tr
 			s_schema.put(key, Arrays.stream(retValue).map(e -> {return new MAcctSchema(Env.getCtx(), e).markImmutable();}).toArray(MAcctSchema[]::new));
 		return retValue;
 	}   //  getClientAcctSchema
+	
+	public static MAcctSchema getSchemaByOrg(Properties ctx, int AD_Client_ID, int AD_Org_ID, String trxName)
+	{
+		MAcctSchema[] list=getClientAcctSchema ( ctx,  AD_Client_ID,  trxName);
+		MAcctSchema general_schema = null;
+		 
+		for (MAcctSchema schema: list)
+		{
+			if (schema.getAD_OrgOnly_ID()==AD_Org_ID)
+				return schema;
+			if (schema.getAD_OrgOnly_ID()==0)
+				general_schema=schema;
+		}
+		 
+		return general_schema;

```

Comment:
This have been talked before and the current definition of acctschema by org is an ambiguous one since the system doesn't stop you from having > 1 acctschema with same ad_orgonly_id.

Reply:
Changed to return Array of schemas. In MPeriod we check if any Org Schema has autoperiod Control, we will use first with autoperiodcontrol set. 

Other solution could be to add Schema_ID as parameter to isOpen method.